airflow/sensors/external_task_sensor.py

```diff
@@ -117,7 +119,9 @@ def poke(self, context, session=None):
         DM = DagModel
         TI = TaskInstance
         DR = DagRun
-        if self.check_existence:
+
+        # we only do the check for 1st time, no need for subsequent poke
+        if self.check_existence and not self.has_check_existence:
```

Comment:
Nothing wrong with this, but with the rescheduling of sensors this won't be persisted :(

Reply:
I think that it is fine as rescheduling interval normally is much longer than poke interval. Currently we load the DagBag using the dag file location in every poke(default  60 sec) if the flag is set to true.